### PR TITLE
Mars photos

### PR DESCRIPTION
**Pull request by liliannaszabo**

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -2,25 +2,86 @@ import express from "express";
 import dotenv from 'dotenv';
 import * as https from "https";
 import axios from "axios";
+import moment from "moment";
 
 dotenv.config();
+enum cameraType{
```
Comment (WendiF) [inline src/server.ts]: Convention - class names, types, etc usually are PascalCase.

File: src/server.ts [hunk 2]
```diff
@@ -2,25 +2,86 @@ import express from "express";
...
 
-app.use(express.json());
+const port = 8000;
 
+app.use(express.json());
 const router = express.Router();
 // router.get('/test', (req: any, res: { send: (arg0: string) => any; }) => res.send());
+
 router.get('/rover', (req: any, res: { send: (arg0: string) => any; }) => {
     axios.get("https://api.nasa.gov/mars-photos/api/v1/rovers?api_key=" + process.env.API_KEY)
```
Comment (WendiF) [inline src/server.ts]: Convention: using template literals

File: src/server.ts [hunk 3]
```diff
@@ -2,25 +2,86 @@ import express from "express";
...
+        dateType = "earth_date";
+        res.send(dateType);
+        return;
+    }
+    else{
+        res.send("You are so shit at URLs");
+        return;
+    }
+
+    axios.get("https://api.nasa.gov/mars-photos/api/v1/rovers/"+req.params.rover + "/photos?sol=1000&camera=" + req.params.camera +"&api_key="+process.env.API_KEY)
```
Comment (WendiF) [inline src/server.ts]: Template literals are particularly valuable here - this is getting a bit hard to follow

File: src/server.ts [hunk 4]
```diff
@@ -2,25 +2,86 @@ import express from "express";
...
 // router.get('/test', (req: any, res: { send: (arg0: string) => any; }) => res.send());
+
 router.get('/rover', (req: any, res: { send: (arg0: string) => any; }) => {
     axios.get("https://api.nasa.gov/mars-photos/api/v1/rovers?api_key=" + process.env.API_KEY)
         .then((r: { data: any; }) => res.send(r.data))
         .catch(function(error){res.send(error)});
 });
 
+
+router.get('/rover/photos', (req:any, res: {send: (arg0:string)=> any}) => {
```
Comment (WendiF) [inline src/server.ts]: What's the reasoning for using `arg0` here?

**Comment by WendiF** [comment]
Have you had any thoughts on how you could refactor this into files yet?

**Review by WendiF — CHANGES_REQUESTED** [review]

File: src/server.ts (hunk 3 above)
Comment (liliannaszabo) [reply src/server.ts]: changed now

File: src/server.ts (hunk 4 above)
Comment (liliannaszabo) [reply src/server.ts]: IDE was being weird, changed now

File: src/server.ts (hunk 1 above)
Comment (liliannaszabo) [reply src/server.ts]: changed now

File: src/server.ts [hunk 5]
```diff
@@ -1,26 +1,42 @@
...
 const router = express.Router();
 // router.get('/test', (req: any, res: { send: (arg0: string) => any; }) => res.send());
-router.get('/rover', (req: any, res: { send: (arg0: string) => any; }) => {
-    axios.get("https://api.nasa.gov/mars-photos/api/v1/rovers?api_key=" + process.env.API_KEY)
-        .then((r: { data: any; }) => res.send(r.data))
-        .catch(function(error){res.send(error)});
-});
+
+
+router.get('/rover/:rover/photos', async (req: any, res:any ) => {
```
Comment (WendiF) [inline src/server.ts]: Can we type these properly? In particular, can we have models for what the response looks like?

File: src/CameraModel.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+
+export type CameraModel = {
+
```
Comment (WendiF) [inline src/CameraModel.ts:3]: Why is this empty?

File: src/PhotoModel.ts [hunk 7]
```diff
@@ -0,0 +1,26 @@
+import {CameraModel} from "./CameraModel";
+import {RoverModel} from "./RoverModel";
+
+
+export type PhotoModel= {
+    id: number,
+    sol: number,
+    camera: {
+        id: number,
```
Comment (WendiF) [inline src/PhotoModel.ts:9]: You've created models for the camera and rover but have instead just added the typings for them in here instead

File: src/server.ts [hunk 8]
```diff
@@ -1,26 +1,41 @@
...
 import axios from "axios";
+import {getRoverPhotosUsingFilters} from "./APIHandler";
+import {PhotoModel} from "./PhotoModel";
 
 dotenv.config();
+
+
+
 const app = express();
-const port = 8000;
```
Comment (WendiF) [inline src/server.ts:14]: Is there a reason for all these empty lines?

File: src/PhotoModel.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    rover: {
+        id: number,
+        name: string,
+        landing_date: string,
+        launch_date: string,
+        status: string,
+    },
+
+};
+
```
Comment (WendiF) [inline src/PhotoModel.ts:26]: These Model/typing files should probably be grouped into a Models folder

File: src/APIHandler.ts [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+export async function getRoverPhotosUsingFilters(rover: Rover, date: string, camera: CameraType)
+{
+    let dateType: string;
+
+    if(date === undefined){
+        date = "1000";
+    }
+    if(camera === undefined){
+        camera = CameraType.fhaz;
```
Comment (WendiF) [inline src/APIHandler.ts]: Have you tried testing what happens when you don't pass these parameters into the url?

File: src/APIHandler.ts [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+
+    if (!(isNaN(Number(date)))) {
+        dateType = "sol"
+    } else if (moment(date, "YYYY-MM-DD").isValid()) {
+        dateType ="earth_date"
+    } else {
+        return {data: "That is not a correct date"};
+    }
```
Comment (WendiF) [inline src/APIHandler.ts:27]: Can you look into what libraries you might be able to use to do response validation?

Comment (WendiF) [reply src/APIHandler.ts:27]: e.g. https://github.com/ianstormtaylor/superstruct
